Odysee/Controllers/Content/SearchViewController.swift

```diff
@@ -152,13 +141,12 @@ class SearchViewController: UIViewController, UIGestureRecognizerDelegate, UISea
     }
     
     func checkNoResults() {
-        DispatchQueue.main.async {
-            self.loadingContainer.isHidden = true
-            self.noResultsView.isHidden = self.claims.count > 0
-            self.noResultsLabel.text = Lighthouse.containsFilteredKeyword(self.currentQuery!) ?
-                String.localized("This search term is disabled to comply with iOS content guidelines. View this search on the web at odysee.com") :
-                String.localized("Oops! We could not find any content matching your search term. Please try again with something different.")
-        }
+        assert(Thread.isMainThread)
+        loadingContainer.isHidden = true
+        noResultsView.isHidden = !claims.isEmpty
+        noResultsLabel.text = Lighthouse.containsFilteredKeyword(currentQuery!) ?
+            String.localized("This search term is disabled to comply with iOS content guidelines. View this search on the web at odysee.com") :
+            String.localized("Oops! We could not find any content matching your search term. Please try again with something different.")
```

Comment:
The reason I remove these dispatches from inside of methods is, it causes problems later on. Dispatching to main causes the block to be run during the _next frame_ of drawing. So if you make some changes to the UI, and then call a method that has a Dispatch inside of it, the changes won't occur at the same time and the user will see a flash or something ugly.

It's better to set the standard: all UIViewController methods are only called on main.